src/main/java/com/microsoft/azure/sample/AppInsightsConfig.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.sample;
+
+import javax.servlet.Filter;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.web.internal.WebRequestTrackingFilter;;
+
+
+@Configuration
+public class AppInsightsConfig {
+
+    @Bean
+    public FilterRegistrationBean aiFilterRegistration() {
+        final FilterRegistrationBean registration = new FilterRegistrationBean();
+        registration.setFilter(new WebRequestTrackingFilter());
+        registration.addUrlPatterns("/**");
+        registration.setOrder(1);
+        return registration;
+    } 
+
+    @Bean(name = "WebRequestTrackingFilter")
+    public Filter WebRequestTrackingFilter() {
+        return new WebRequestTrackingFilter();
```

Comment:
Well have you tried to instrument Application Insights Agent to collect dependencies in this application? Probably if you would like to collect the calls to documentDB (unfortunately not autocollected now) or other dependencies like HTTP outgoing via Apache HTTP, Redis etc. you would need to put a "name of type string representing application name" as a parameter to WebRequestTrackingFilter() . Feel free to ask for more details

Reply:
As we discussed, there is a bug that is blocking this - I logged an issue to add dependency tracking when the bug is fixed: https://github.com/Microsoft/todo-app-java-on-azure/issues/36